jQuery/ajax.js

```diff
@@ -1,82 +1,113 @@
+
+  function redrawContent(contentSelector, citizens) {
+    var content = $(contentSelector);
+    content.empty();
+
+    citizens.forEach(function(citizen) {
+      content.append($('<div>', { class: "citizen", html: citizen.name }));
+    });
+  }
+
+  return {
+    gatherParams: gatherParams,
+    populateSelectWithOptions: populateSelectWithOptions,
+    redrawContent: redrawContent
   };
+})();
+
+/* Defines funtions that handle backend logic that handle data, nothing to do with DOM */
+window.hh.logic = window.hh.logic || (function() {
+  function getUniqueValuesForField(field, citizens) {
+    var values = ['All'];
+    citizens.forEach(function(c) {
+      var v = c[field];
+      if (values.indexOf(v) === -1) {
+        values.push(v);
+      }
+    });
+    return values;
+  }
+
+  function filter(params, citizens) {
+    return citizens.filter(function(citizen) {
+      var matchesFilter = true;
+
+      for (var i = 0; i < params.length && matchesFilter; i++) {
+        var param = params[i];
+
+        var field = param['field'];
+        var citizenValue = citizen[field];
+        var paramValue = param['value'];
+
+        matchesFilter = paramValue === 'All' || citizenValue.toUpperCase().indexOf(paramValue.toUpperCase()) !== -1;
+      }
+      return matchesFilter;
+    });
+  }
+
+  return {
+    getUniqueValuesForField: getUniqueValuesForField,
+    filter: filter
+  };
+})();
 
-  $('#character-select').on('change', { citizens: citizens }, selectHandler);
-  $('#hometown-select').on('change', { citizens: citizens }, selectHandler);
-  $('#size-select').on('change', { citizens: citizens }, selectHandler);
-
-  selectHandler({data: { citizens: citizens }});
-}
-
-function getUniqueValuesForField(field, citizens) {
-  var values = ['All'];
-  citizens.forEach(function(c) {
-    var v = c[field];
-    if (values.indexOf(v) === -1) {
-      values.push(v);
-    }
-  });
-  return values;
-}
-
-function populateSelectWithOptions(select, options) {
-  options.forEach(function(option) {
-    $(select).append($('<option>', { value: option, html: option }));
-  });
-}
-
-function gatherParams() {
-  var params = [];
-
-  var characterValue = $('#character-select').val();
-  params.push({field: 'character', value: characterValue});
-
-  var hometownValue = $('#hometown-select').val();
-  params.push({field: 'hometown', value: hometownValue});
-
-  var sizeValue = $('#size-select').val();
-  params.push({field: 'size', value: sizeValue});
-
-  return params;
-}
-
-function filter(params, citizens) {
-  var filteredCitizens = citizens.filter(function(citizen) {
-    var matchesFilter = true;
-
-    for (var i = 0; i < params.length && matchesFilter; i++) {
-      var param = params[i];
-
-      var field = param['field'];
-      var citizenValue = citizen[field];
-      var paramValue = param['value'];
+window.hh.ajax = window.hh.ajax || function(citizens) {
+  var selectors = window.hh.selectors;
+  var DOM = window.hh.domManipulation;
```

Comment:
So this is where we get into _dependency injection_ with modules. You can [read this](https://morklympious.github.io/dependency-injection-what/) if you want 

it's a bad idea to declare this explicitly in the module. Instead, we want these module patterns to work for us, so the idea is to pass in the thing you're going to use. For example, this is a very common pattern for a module using jQuery:

```js
window.utilities = window.utilities || (function($) {

...

})(jQuery)
```
Notice how I passed in the global reference to jQuery so it populates the parameter I defined as `$`? You'll want to do the same with your `ajax` module. 

e.g.

```js
(function(citizens, logic) {

// I can now use window.hh.logic's API, available through the logic param
// No declaration needed. 

}( /*whatever citizens is */, window.hh.logic));
```